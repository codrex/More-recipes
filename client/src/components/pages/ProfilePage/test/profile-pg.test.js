import React from 'react';
import { Provider } from 'react-redux';
import { ProfilePage } from '../index';
import configureStore from '../../../../store/configStore';
import toJson from 'enzyme-to-json';

const store = configureStore();
const props = {
  actions: {
    getProfile: jest.fn(),
    update: jest.fn(),
    currentRecipe: jest.fn(),
    deleteRecipe: jest.fn(),
    removeFromFav: jest.fn(),
    resetSuccess: jest.fn(),

  },
  match: { history: { push: jest.fn() } },
  loading: false,
  success: false,
  user: {
    favRecipes: [{
      name: 'recipe',
      category: '',
      views: 1,
      upVotes: 1,
      downVotes: 1,
      recipeId: 1,
    }],
    createdRecipes: [{
      name: 'recipe',
      category: '',
      views: 1,
      upVotes: 1,
      downVotes: 1,
      recipeId: 1,
    }],
  },
};

describe('Profile Page component ', () => {
  test('to match empty snapshot ', () => {
    const wrapper = shallow(<ProfilePage {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
  test('render as expected when edit profile button is clicked ', () => {
    const wrapper = mount(
      <Provider store={store}>
        <ProfilePage {...props} />
      </Provider>
    );
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    wrapper.find('UserInfo').find('#editProfile').simulate('click');
    expect(tree).toMatchSnapshot();
  });
});

