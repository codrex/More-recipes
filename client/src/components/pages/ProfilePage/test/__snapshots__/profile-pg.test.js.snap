// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page component  render as expected when edit profile button is clicked  1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ProfilePage
    actions={
      Object {
        "currentRecipe": [Function],
        "deleteRecipe": [Function],
        "getCreatedRecipes": [Function],
        "getProfile": [Function],
        "removeFromFav": [Function],
        "resetSuccess": [Function],
        "update": [Function],
      }
    }
    history={
      Object {
        "push": [Function],
      }
    }
    loading={false}
    pageCount={0}
    recipes={Array []}
    success={false}
    user={
      Object {
        "createdRecipes": Array [
          Object {
            "category": "",
            "downVotes": 1,
            "name": "recipe",
            "recipeId": 1,
            "upVotes": 1,
            "views": 1,
          },
        ],
        "favRecipes": Array [
          Object {
            "category": "",
            "downVotes": 1,
            "name": "recipe",
            "recipeId": 1,
            "upVotes": 1,
            "views": 1,
          },
        ],
      }
    }
  >
    <div
      className="container-fluid profile-page no-padding"
    >
      <HeroArea
        image="https://res.cloudinary.com/resycom/image/upload/c_scale,h_728,q_51/v1510430032/lily-lvnatikk-365344_nv94dc.jpg"
        title=""
      >
        <div
          className="hero-half backdrop"
        >
          <img
            alt="heroimage"
            src="https://res.cloudinary.com/resycom/image/upload/c_scale,h_728,q_51/v1510430032/lily-lvnatikk-365344_nv94dc.jpg"
          />
          <h1
            className="display-4
          text-capitalize
          col-8
          bold
          text-center
          "
            id="display-1"
          >
            <UserInfo
              editBtnClicked={[Function]}
              user={
                Object {
                  "createdRecipes": Array [
                    Object {
                      "category": "",
                      "downVotes": 1,
                      "name": "recipe",
                      "recipeId": 1,
                      "upVotes": 1,
                      "views": 1,
                    },
                  ],
                  "favRecipes": Array [
                    Object {
                      "category": "",
                      "downVotes": 1,
                      "name": "recipe",
                      "recipeId": 1,
                      "upVotes": 1,
                      "views": 1,
                    },
                  ],
                }
              }
            >
              <div
                className="info-wrapper mr-auto d-flex flex-column col-xs-12 col-sm-10 col-md-8 col-lg-6"
              >
                <div
                  className=" d-flex justify-content-around flex-column align-items-center"
                >
                  <div
                    className="avatar avatar-md"
                  />
                  <h3
                    className=" text-capitalize s-padding no-margin "
                  />
                </div>
                <div
                  className="d-flex no-margin flex-column align-items-center justify-content-center info"
                >
                  <li
                    className="info-item"
                  />
                  <li
                    className="info-item"
                  >
                    @undefined
                  </li>
                </div>
                <Button
                  Tag="button"
                  className=" btn-lg center-margin edit-btn btn-secondary"
                  dataTarget="#modal"
                  dataToggle="modal"
                  handleClick={[Function]}
                  href=""
                  id="editProfile"
                  text="edit profile"
                  type="button"
                >
                  <button
                    className="btn  btn-lg center-margin edit-btn btn-secondary"
                    data-dismiss="modal"
                    data-target="#modal"
                    data-toggle="modal"
                    href=""
                    onClick={[Function]}
                    type="button"
                  >
                    edit profile
                  </button>
                </Button>
              </div>
            </UserInfo>
          </h1>
        </div>
      </HeroArea>
      <NotFound
        message="you have no recipes"
      >
        <div
          className="display-3 text-uppercase msg-display no-margin d-flex flex-column "
        >
          <h1
            className="not-found-text"
          >
            <i
              className="fa fa-meh-o"
            />
            <em>
              oops!
            </em>
            <br />
            you have no recipes
          </h1>
          <Button
            Tag="button"
            className="btn-secondary-outline btn-lg"
            dataTarget=""
            dataToggle=""
            handleClick={[Function]}
            href=""
            text="add recipe"
            type="button"
          >
            <button
              className="btn btn-secondary-outline btn-lg"
              data-dismiss="modal"
              data-target=""
              data-toggle=""
              href=""
              onClick={[Function]}
              type="button"
            >
              add recipe
            </button>
          </Button>
        </div>
      </NotFound>
      <Modal
        center={true}
        closeBtnClicked={[Function]}
        footer={false}
        id="editProfileModal"
        left={true}
        leftBtnText=""
        loading={false}
        onContinueClicked={[Function]}
        operationCompleted={false}
        right={true}
        rightBtnText=""
        title=""
      >
        <div
          aria-hidden="true"
          aria-labelledby="editProfileModalLabel"
          className="modal fade"
          data-backdrop="false"
          data-show="true"
          id="modal"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content col-xs-12 col-sm-12 col-md-8 col-lg-5"
            >
              <div
                className="modal-header"
              >
                <h5
                  className="modal-title text-capitalize display-4"
                  id="editProfileModalLabel"
                />
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    Ã—
                  </span>
                </button>
              </div>
              <div
                className="modal-body"
              />
            </div>
          </div>
        </div>
      </Modal>
    </div>
  </ProfilePage>
</Provider>
`;

exports[`Profile Page component  render as expected when edit profile button is clicked  2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ProfilePage
    actions={
      Object {
        "currentRecipe": [Function],
        "deleteRecipe": [Function],
        "getCreatedRecipes": [Function],
        "getProfile": [Function],
        "removeFromFav": [Function],
        "resetSuccess": [Function],
        "update": [Function],
      }
    }
    history={
      Object {
        "push": [Function],
      }
    }
    loading={false}
    pageCount={0}
    recipes={Array []}
    success={false}
    user={
      Object {
        "createdRecipes": Array [
          Object {
            "category": "",
            "downVotes": 1,
            "name": "recipe",
            "recipeId": 1,
            "upVotes": 1,
            "views": 1,
          },
        ],
        "favRecipes": Array [
          Object {
            "category": "",
            "downVotes": 1,
            "name": "recipe",
            "recipeId": 1,
            "upVotes": 1,
            "views": 1,
          },
        ],
      }
    }
  >
    <div
      className="container-fluid profile-page no-padding"
    >
      <HeroArea
        image="https://res.cloudinary.com/resycom/image/upload/c_scale,h_728,q_51/v1510430032/lily-lvnatikk-365344_nv94dc.jpg"
        title=""
      >
        <div
          className="hero-half backdrop"
        >
          <img
            alt="heroimage"
            src="https://res.cloudinary.com/resycom/image/upload/c_scale,h_728,q_51/v1510430032/lily-lvnatikk-365344_nv94dc.jpg"
          />
          <h1
            className="display-4
          text-capitalize
          col-8
          bold
          text-center
          "
            id="display-1"
          >
            <UserInfo
              editBtnClicked={[Function]}
              user={
                Object {
                  "createdRecipes": Array [
                    Object {
                      "category": "",
                      "downVotes": 1,
                      "name": "recipe",
                      "recipeId": 1,
                      "upVotes": 1,
                      "views": 1,
                    },
                  ],
                  "favRecipes": Array [
                    Object {
                      "category": "",
                      "downVotes": 1,
                      "name": "recipe",
                      "recipeId": 1,
                      "upVotes": 1,
                      "views": 1,
                    },
                  ],
                }
              }
            >
              <div
                className="info-wrapper mr-auto d-flex flex-column col-xs-12 col-sm-10 col-md-8 col-lg-6"
              >
                <div
                  className=" d-flex justify-content-around flex-column align-items-center"
                >
                  <div
                    className="avatar avatar-md"
                  />
                  <h3
                    className=" text-capitalize s-padding no-margin "
                  />
                </div>
                <div
                  className="d-flex no-margin flex-column align-items-center justify-content-center info"
                >
                  <li
                    className="info-item"
                  />
                  <li
                    className="info-item"
                  >
                    @undefined
                  </li>
                </div>
                <Button
                  Tag="button"
                  className=" btn-lg center-margin edit-btn btn-secondary"
                  dataTarget="#modal"
                  dataToggle="modal"
                  handleClick={[Function]}
                  href=""
                  id="editProfile"
                  text="edit profile"
                  type="button"
                >
                  <button
                    className="btn  btn-lg center-margin edit-btn btn-secondary"
                    data-dismiss="modal"
                    data-target="#modal"
                    data-toggle="modal"
                    href=""
                    onClick={[Function]}
                    type="button"
                  >
                    edit profile
                  </button>
                </Button>
              </div>
            </UserInfo>
          </h1>
        </div>
      </HeroArea>
      <NotFound
        message="you have no recipes"
      >
        <div
          className="display-3 text-uppercase msg-display no-margin d-flex flex-column "
        >
          <h1
            className="not-found-text"
          >
            <i
              className="fa fa-meh-o"
            />
            <em>
              oops!
            </em>
            <br />
            you have no recipes
          </h1>
          <Button
            Tag="button"
            className="btn-secondary-outline btn-lg"
            dataTarget=""
            dataToggle=""
            handleClick={[Function]}
            href=""
            text="add recipe"
            type="button"
          >
            <button
              className="btn btn-secondary-outline btn-lg"
              data-dismiss="modal"
              data-target=""
              data-toggle=""
              href=""
              onClick={[Function]}
              type="button"
            >
              add recipe
            </button>
          </Button>
        </div>
      </NotFound>
      <Modal
        center={true}
        closeBtnClicked={[Function]}
        footer={false}
        id="editProfileModal"
        left={true}
        leftBtnText=""
        loading={false}
        onContinueClicked={[Function]}
        operationCompleted={false}
        right={true}
        rightBtnText=""
        title=""
      >
        <div
          aria-hidden="true"
          aria-labelledby="editProfileModalLabel"
          className="modal fade"
          data-backdrop="false"
          data-show="true"
          id="modal"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content col-xs-12 col-sm-12 col-md-8 col-lg-5"
            >
              <div
                className="modal-header"
              >
                <h5
                  className="modal-title text-capitalize display-4"
                  id="editProfileModalLabel"
                />
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    Ã—
                  </span>
                </button>
              </div>
              <div
                className="modal-body"
              />
            </div>
          </div>
        </div>
      </Modal>
    </div>
  </ProfilePage>
</Provider>
`;

exports[`Profile Page component  to match empty snapshot  1`] = `
<div
  className="container-fluid profile-page no-padding"
>
  <HeroArea
    image="https://res.cloudinary.com/resycom/image/upload/c_scale,h_728,q_51/v1510430032/lily-lvnatikk-365344_nv94dc.jpg"
    title=""
  >
    <UserInfo
      editBtnClicked={[Function]}
      user={
        Object {
          "createdRecipes": Array [
            Object {
              "category": "",
              "downVotes": 1,
              "name": "recipe",
              "recipeId": 1,
              "upVotes": 1,
              "views": 1,
            },
          ],
          "favRecipes": Array [
            Object {
              "category": "",
              "downVotes": 1,
              "name": "recipe",
              "recipeId": 1,
              "upVotes": 1,
              "views": 1,
            },
          ],
        }
      }
    />
  </HeroArea>
  <NotFound
    message="you have no recipes"
  >
    <Button
      Tag="button"
      className="btn-secondary-outline btn-lg"
      dataTarget=""
      dataToggle=""
      handleClick={[Function]}
      href=""
      text="add recipe"
      type="button"
    />
  </NotFound>
  <Modal
    center={true}
    closeBtnClicked={[Function]}
    footer={false}
    id="editProfileModal"
    left={true}
    leftBtnText=""
    loading={false}
    onContinueClicked={[Function]}
    operationCompleted={false}
    right={true}
    rightBtnText=""
    title=""
  />
</div>
`;
